client.go

```diff
@@ -180,7 +180,26 @@ func runStreamedCommand(img, cmd string, getError bool) error {
 // It returns the ID of the container spawned from the image.
 // Note well: the container is NOT deleted when the given command terminates.
 func runCommandInContainer(img string, cmd []string, dst io.Writer) (string, error) {
-	return startContainer(img, cmd, true, dst)
+	id, err := createContainer(img, cmd)
+	if err != nil {
+		log.Println(err)
+		return "", err
+	}
+
+run_cmd:
+	id, err = startContainer(id, cmd, true, dst)
+	switch err.(type) {
+	case dockerError:
+		de := err.(dockerError)
+		if isZypperExitCodeSevere(de.exitCode) {
+			if de.exitCode == zypperExitInfRestartNeeded {
+				goto run_cmd
```

Comment:
The nesting of those if-branches isn't correct here. First, we need to return an error if the exit code is severe. Also the zypperExitInfRestartNeeded has the value 103, but severe ones are `> 0 && < 100`, so the if conditions are contradicting. What we should do instead is to return an error if the code is severe and move the run_cmd in a separate if branch.